### ISSUE TITLE
Xdrip+ with Blucon Nightrider on Full-Android-Smartwatch (No.1 D6, Android 5.1): Bluetooth issue

### ISSUE THREAD

**Issue by dideldum** [NightscoutFoundation/xDrip#216]
I installed Xdrip+ on a smartwatch with normal Android 5.1 (No.1 D6).
I first had a lot of problems connecting Xdrip+ to the Blucon Nightrider, I got many times the error "Invalid PIN", though the PIN definitely was entered correctly.
I had the same problem also on a smartphone (Elephone P8000) with Android 5.1.
Somehow finally I got the connection to the Nightrider. All works fine, until the Nightrider leaves the Bluetooth range of the watch.
When in range again, sometimes XDrip+ reconnects to the Blucon, but very often it doesn't. Even after 30min and switching Bluetooth off and on and doing "Restart Collector" there is no reconnection. Rebooting the watch and/or switching the Blucon off and on normally make the system working again.
On a tablet with Android 7.0 XDrip+ always reconnects within 1-3mins normally without any further action.
Is there an issue with Android 5.1/Bluetooth?
I'm very interested to get this smartwatch working, as it would make it easier to carry it as a master than it is with the tablet.

**Comment by jamorham** [comment]
Which xDrip+ version are you using?

**Comment by dideldum** [comment]
I tried various nightly builds from end of september to november. Currently the version from 2017-10-26 is installed, as I had it installed on all other devices. The latest version I tried must have been around Nov 10th.
But it seems, it doesn't have to do with Android 5.1. In a facebook group someone wrote, he/she had a reconnection issue for 5 hours over night with an Android 7.0 device.
I would like to look at the logs, but I cannot read them on the small smartwatch display. Is there a way to extract the log as a file? I have root access to the smartwatch, so it's not a problem to access any directory.

**Comment by dideldum** [comment]
Have you changed something with Bluetooth?
I installed Nightly Build 2017-11-29 and had 2 times a perfect reconnect within 2 minutes, once after 12mins and once after 20mins of big distance to Blucon.
*Confused but happy*

**Comment by dideldum** [comment]
... No, unfortunately still no reliable reconnection.
Worked two times, now suddenly no reconnection after 20mins, even not when Bluetooth is disabled/enabled, Restart Collector is done (both with Bluetooth disabled and enabled)...
So even in 2017-11-29 not working.

**Comment by dideldum** [comment]
Hm, maybe it's not an issue with XDrip+ at all.
I disabled GPS to save energy. Since then even rebooting the watch didn't help to reconnect. 
After enabling GPS it immediately connected.
I'll test it more intensively.

**Comment by gregorybel** [comment]
Can you provide logs?

Are you sure you removed original app or glimp?

**Comment by dideldum** [comment]
Other apps than XDrip+ were never installed.
Log was just sent from smartwatch.

**Comment by masipmarch** [comment]
Sorry @dideldum have you solved the issue? I've the same issue with another smartwatch, my smartwatch is an Ourtime X01S, with Android 5.1 too. I've tried a lot of combinations but without lucky. I only get connection with a bq phone with andriod 6.

**Comment by dideldum** [comment]
@masipmarch I have just received my replaced Ourtime X01 Air 2 days ago, but so far I could not test with my daughter's Blucon. Her Blucon had more or less stopped working, the batteries were drained in 16 hours, so I had sent it to Ambrosia. I'm waiting for another Blucon sent via my American cousin  to avoid customs problems in Germany.
With the No.1 D6 I had not managed to get a reconnection with the Blucon in xDrip+. As long as the watch stayed in Bluetooth range, it was not a problem, but leaving the Bluetooth range left it disconnected (always in "Connecting" state, watch had to be rebooted to connect again).
As the Blucon was not working anymore and the Sony Smartwatch 3 is very uncomfortable for my 4 years old daughter, I bought an NFC capable smartphone (Blackview BV6000) as an intermediate solution, which is currently used with the SSW3 at night and to scan directly during the day.
So at the moment I am on hold, until we have a new Blucon. Then I will test with the Ourtime X01 Air.
But I can tell you, that a greek friend has the Microwear H5 for a few days now and is enthusiastic about it. Good connection to Blucon (though I don't know, if he tested the reconnection), very good battery (650mAh!), works perfect as a master.
I'm considering to buy the Microwear H5, too.

**Comment by masipmarch** [comment]
Thanks for  response @dideldum , now i've my Ourtime out of battery, i'll
try later to do more tests, doing a reset and triying with GPS active. It's
for my 7 years old daughter, and i think the Ourtime it's too big in her
arm, we'll see. My idea it's that she carry only the smartwatch with xdrip+
and send data to nightscout. If i get Ourtime working in a good conditions
i'll tell you.

Thanks for tell me about Microwear H5 , it's smaller than Ourtime and have
a good battery...i think it's a good idea if i can't get Ourtime working
correctly.

I'm in ha facebook group about nightscout in spain, where there're some
people that have it working with a smarwatch zgpax s8, but it's bigger and
with only 450 mAh of battery.

Maybe the problem it's in android version because zgpax s8 have android 4.4
and it works, Microwear H5 have android 6 and it works, and Ourtime have
android 5.1 like No.1 D6 and have problems with BT connection.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by dideldum** [comment]
If my Ourtime works, I will try to get the antennas out of the arm wrist and stick them somehow at the bottom of the watch. Then the watch could be worn by our daughter in another small pocket in her cloth belt (sorry for a probably bad translation from german word "Bauchband") beside her pump. Thus it should not bother her too much.

**Comment by masipmarch** [comment]
I get it working but only for some minutes. i paired blucon with smartwatch
BT and then connected it with xdrip+ and it works, but only 60-70 minutes,
it desconnect's automatically...and needs to  restart smartwatch, connect
to smartwatch BT, and then connect to xdrip+ another time. The smartwatch
it's all the time in the same wirst where the blucon is.I'll try to do more
tests this afternoon.

I think its a good idea getting out the antennas from arm wirst for a child
and put in a small pocket.

Do not worry about the translation, mine is checked for google translate ;-)


Here when it works about a hour....




and then stopped working... :-(


​​

**Comment by dideldum** [comment]
@masipmarch Possibly you are right with your hint, that Android 5.1 may cause the problem. I tried also to connect a smartphone with Android 5.1 to the Blucon via xDrip+. It failed with "invalid PIN" while trying to connect to the Blucon, but it is possible, that this error came, because I didn't remove the Blucon before from the normally used tablet, I just switched the tablet off. So I am not sure, if the error was really Android 5.1 related.
